### ISSUE TITLE
I want to build wasm for Deno.

### ISSUE THREAD

**Issue by kahirokunn** [rustwasm/wasm-pack#672]
## 💡 Feature description
I want to build wasm for Deno.
https://github.com/denoland/deno

I try write rust and do wasm-pack.
And then I got .wasm file, I put it file on the Deno.
But the wasm file never work....(´；ω；｀)

#### 💻 Basic example

```sh
$ wasm-pack build --target deno
```

**Comment by csmoe** [comment]
`wasm-pack`'s build target reiles on `wasm-bindgen` cc https://rustwasm.github.io/docs/wasm-bindgen/reference/cli.html?highlight=target#--target

**Comment by ibaryshnikov** [comment]
@kahirokunn is there `require` function in deno?

**Comment by ashleygwilliams** [comment]
related https://github.com/denoland/deno/issues/1325

**Comment by ashleygwilliams** [comment]
@kahirokunn in order to run wasm-packed wasm, you'll need to require the js wrapper generated by wasm-bindgen. can deno load JS?

**Comment by kevinkassimo** [comment]
fyi Deno has resolved the original issue posted above and have recently just introduced WASM imports (should be very similar to Node.js's `--experimental-wasm-modules`):
```ts
import { exportedFunc1, exportedFunc2 } from "./my.wasm";
```
If `my.wasm` imports from other JS/TS files, it could be done by specifying import names like
```wat
(module
  (import "./wasm-dep.js" "jsFn" (func $jsFn (result i32)))
  (import "http://127.0.0.1:4545/cli/tests/051_wasm_import/remote.ts" "jsRemoteFn" (func $jsRemoteFn (result i32)))
)
```
with corresponding local/remote file set up.

**Comment by ytoune** [comment]
I use below for a while

```sh
cargo generate --git https://github.com/rustwasm/wasm-pack-template -n myproject
cd myproject
# deno has not alert but has console.log.
sed -i -e "s/fn alert(s: &str);/#[wasm_bindgen(js_namespace = console, js_name = log)] fn alert(s: \\&str);/" src/lib.rs

wasm-pack build --target web --out-name index
# deno hasn't supported fetch('file:// ...') yet.
sed -i -e "s#input = import\\.meta\\.url\\.replace(/\\\\\\.js$/, '_bg\\.wasm');#input = import.meta.url.replace(/\\\\.js$/, '_bg.wasm'); if ('undefined' !== typeof Deno) input = new WebAssembly.Module(await Deno.readFile(new URL(input).pathname));#" pkg/index.js

echo "import * as pkg from './pkg/index.js'
await pkg.default()
pkg.greet()" >| app.js
deno run --allow-read app.js
```

https://github.com/denoland/deno/issues/2150

**Comment by bsunderhus** [comment]
Just for future references of those who are still a little confused and end up in this opened issue. 

> To build for deno just build for web: `wasm-pack build --target web`

The only problem at the moment is that [Deno doesn't support fetch for local files](https://github.com/denoland/deno/issues/2150), and the web targeted build will try to get the `.wasm` binary with **fetch**, so to solve this when calling the initializing function from the generated package you'll have to manually pass the content extracted from the `.wasm` file

example:

```ts
import init, {greet} from "./pkg/greet.js";
await init()
greet()
```

**Comment by drager** [comment]
Fixed in [v0.11.0](https://github.com/rustwasm/wasm-pack/releases/tag/v0.11.0).
